src/sentry/incidents/logic.py

```diff
@@ -854,6 +856,12 @@ def update_alert_rule(
             updated_fields["seasonality"] = None
         elif detection_type == AlertRuleDetectionType.DYNAMIC:
             updated_fields["comparison_delta"] = None
+            if (
+                time_window not in {15, 30, 60}
+                or time_window is None
+                and alert_rule.snuba_query.time_window not in {15 * 60, 30 * 60, 60 * 60}
+            ):
+                raise ValidationError("Invalid time window for dynamic alert")
```

Comment:
tiny nit but since the validation error strings are repeated we could put them in a const to avoid potential future drift